File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/Functions/AbsFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions
{
    public class AbsFunction : SingleArgumentFunction
    {
        public const string Name = "ABS";

        protected override string FunctionHelp => $"{Name} expects 1 arguments (Value)";

        protected override VariableType ArgumentType => PrimitiveType.Number;
        protected override VariableType ResultType => PrimitiveType.Number;

        private AbsFunction(Expression valueExpression, SourceReference reference)
            : base(valueExpression, reference)
        {
        }

        public static BuiltInFunction Create(SourceReference reference, Expression expression) =>
            new AbsFunction(expression, reference);
    }
}

File: typescipt/src/language/expressions/functions/absFunction.ts


export class AbsFunction extends SingleArgumentFunction {
   public const string Name = `ABS`;

   protected override string FunctionHelp => $`{Name} expects 1 argument (Value)`;

   protected override VariableType ArgumentType => PrimitiveType.number;
   protected override VariableType ResultType => PrimitiveType.number;

   constructor(valueExpression: Expression, reference: SourceReference)
     : base(valueExpression, reference) {
   }

   public static create(reference: SourceReference, expression: Expression): ExpressionFunction {
     return new AbsFunction(expression, reference);
   }
}
